Mathlib/Algebra/Module/Submodule/Basic.lean

```diff
@@ -365,29 +351,6 @@ instance noZeroSMulDivisors [NoZeroSMulDivisors R M] : NoZeroSMulDivisors R p :=
     this.imp_right (@Subtype.ext_iff _ _ x 0).mpr⟩
 #align submodule.no_zero_smul_divisors Submodule.noZeroSMulDivisors
 
-/-- Embedding of a submodule `p` to the ambient space `M`. -/
-protected def subtype : p →ₗ[R] M := by refine' { toFun := Subtype.val.. } <;> simp [coe_smul]
-#align submodule.subtype Submodule.subtype
```

Comment:
I'm not convinced by this split; for `Submonoid` and `Subgroup`, don't we define this map pretty immediately?

Reply:
No for `Submonoid` (it's in `Mathlib.GroupTheory.Submonoid.Operations`) but yes for `Subgroup` (in `Mathlib.GroupTheory.Subgroup.Basic`. I felt this file had pretty heavy imports, but I'm also happy to revert this part.

Reply:
@Vierkantor, I'll let you make the call on this

Reply:
My impression is that `Submodule.subtype` is not used too commonly, so I don't mind the split. This also means we don't have to import `LinearMap` when defining `Submodule` which is nice for combing the imports.